timur/lib/client/jsx/components/timur_nav.jsx

```diff
@@ -2,120 +2,64 @@
 import * as React from 'react';
 import {connect} from 'react-redux';
 
+import Nav from 'etna-js/components/Nav';
 import IdentifierSearch from './identifier_search';
 import Link from './link';
-import {selectUser} from '../selectors/user_selector';
+import {selectUser} from 'etna-js/selectors/user-selector';
+import {useReduxState} from 'etna-js/hooks/useReduxState';
 
-const Halo = ({radius}) => (
+const Halo = ({radius}) =>
   <div className="halo">
     <svg>
       <circle r={`${radius * 0.7}px`} cx={`${radius}px`} cy={`${radius}px`} />
-      {Array(36)
-        .fill()
-        .map((_, i) => {
+      {Array(36).fill().map((_, i) => {
           let deg = i * 10;
           let rad = radius * (i % 2 == 0 ? 1.0 : 0.9);
           let x = (r) => Math.cos((Math.PI * deg) / 180) * r + radius;
           let y = (r) => Math.sin((Math.PI * deg) / 180) * r + radius;
-          return (
-            <path
-              className={i % 2 == 0 ? 'long' : 'short'}
-              key={i}
-              d={`M ${x(rad)}, ${y(rad)} L ${x(radius * 0.7)}, ${y(
-                radius * 0.7
-              )}`}
-            />
-          );
+          return <path key={i}
+            className={i % 2 == 0 ? 'long' : 'short'}
+            d={`M ${x(rad)} ${y(rad)} L ${x(radius * 0.7)} ${y(radius * 0.7)}`}
+          />;
         })}
     </svg>
-  </div>
-);
+  </div>;
 
-const Logo = ({exchanges, logo_id}) => (
-  <div id="logo">
-    <a href="/">
-      <div
-        id={logo_id}
-        className={Object.keys(exchanges).length > 0 ? 'throb' : null}
-      >
-        <div className="image"></div>
-        <Halo radius={27.5} />
-      </div>
-    </a>
+const Logo = connect(({exchanges}) => ({exchanges}))(({exchanges}) =>
+  <div id='timur-logo'
```

Comment:
I ended up using `connect` here because `let exchanges = useReduxState(state=>state.exchanges)` did not behave equivalently for reasons I don't quite understand.

Reply:
It may be behavior difference for arrays vs objects in change tracking, unfortunately.  I'd have to take a closer look.